### DISCUSSION TITLE
Parsing prefixed radix numbers

### DISCUSSION THREAD

**Discussion by mainrs** [zesterer/chumsky#244] [Q&A] [answered]
I am trying to write a parser for lisp-like number representations. These are `#nnRdddd` where `n` is a digit representing the radix and `d` is a digit in said radix. I want the parser to first parse the radix and then for that given radix parse the string of digits `dddd`. However, when composing parsers, I cannot partially evaluate the radix parser first and use its result for the digit parser.

I do not really differentiate between lexer and parser. I directly parse into an AST.

What approaches can I take here?

**Comment by zesterer** [comment, marked as answer]
You can currently do this with the [`then_with`](https://docs.rs/chumsky/latest/chumsky/trait.Parser.html#method.then_with) combinator, but it's something that's likely going to be removed in a future version of the crate in favour of some other mechanism for customising the behaviour of future parsers based on the output of previous ones.

Alternatively, you can just parse add of the `#nnRdddd` with all possible digits and use `try_map` to do whatever manual consistency checks you want inline.

**Reply by mainrs** [reply]
Thank you very much! I forgot about `then_with`.
